includes/core/classes/blocks/class-event-query.php

```diff
@@ -0,0 +1,266 @@
+	 * @param array       $parsed_block The block being rendered.
+	 * @return string|null The pre-rendered content. Default null.
+	 */
+	public function pre_render_block( ?string $pre_render, array $parsed_block ): ?string {
+		if ( isset( $parsed_block['attrs']['namespace'] ) && 'gatherpress-event-query' === $parsed_block['attrs']['namespace'] ) {
+			if ( isset( $parsed_block['attrs']['query']['inherit'] ) && true === $parsed_block['attrs']['query']['inherit'] ) {
+				global $wp_query;
+				$query_args = array_merge(
+					$wp_query->query_vars,
+					array(
+						'posts_per_page' => $parsed_block['attrs']['query']['perPage'],
+						'order'          => $parsed_block['attrs']['query']['order'],
+						'orderby'        => $parsed_block['attrs']['query']['orderBy'],
+					)
+				);
+
+				/**
+				 * Filter the query vars.
+				 *
+				 * Allows filtering query params when the query is being inherited.
+				 *
+				 * @since 1.5
+				 *
+				 * @param array   $query_args  Arguments to be passed to WP_Query.
+				 * @param array   $block_query The query attribute retrieved from the block.
+				 * @param boolean $inherited   Whether the query is being inherited.
+				 *
+				 * @return array $filtered_query_args Final arguments list.
+				 */
+				$filtered_query_args = \apply_filters(
+					'gatherpress_query_vars',
+					$query_args,
+					$parsed_block['attrs']['query'],
+					true,
+				);
+				// "Hijack the global query. It's a hack, but it works." Ryan Welcher
+				$wp_query = new WP_Query( $filtered_query_args ); // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+			} else {
+				add_filter(
+					'query_loop_block_query_vars',
+					array( $this, 'query_loop_block_query_vars' ),
+					10,
+					2
+				);
+			}
+		}
+
+		return $pre_render;
+	}
+
+	/**
+	 * Returns an array with Post IDs that should be excluded from the Query.
+	 *
+	 * @param array $attributes Event Query block attributes.
+	 * @return int[] Array of post IDs to exclude.
+	 */
+	protected function get_exclude_ids( array $attributes ): array {
+		$exclude_ids = array();
+
+		// Exclude Current Post.
+		if ( isset( $attributes['exclude_current'] ) && boolval( $attributes['exclude_current'] ) ) {
+			array_push( $exclude_ids, $attributes['exclude_current'] );
+		}
+
+		return $exclude_ids;
+	}
+
+	/**
+	 * Filters the arguments which will be passed to `WP_Query` for the Query Loop Block.
+	 *
+	 * @param array     $query Array containing parameters for <code>WP_Query</code> as parsed by the block context.
+	 * @param \WP_Block $block Block instance.
+	 * @return array Array containing parameters for <code>WP_Query</code> as parsed by the block context.
+	 */
+	public function query_loop_block_query_vars( array $query, \WP_Block $block ): array {
+		// Retrieve the query from the passed block context.
+		$block_query = $block->context['query'];
+
+		if ( ! is_array( $block_query ) || ! isset( $block_query['gatherpress_events_query'] ) ) {
```

Comment:
Thinking maybe add a constant on the class for `gatherpress_events_query`. See other block classes.

Reply:
I added a const for 'gatherpress-event-query' as BLOCK_NAME (like in other block classes).

But this here relates to the query param, which has no class constant (yet).
Mention the plural in 'gatherpress_events_query'.